text/0000-stable-wakers.md

```diff
@@ -0,0 +1,196 @@
+- Feature Name: `stable_wakers`
+- Start Date: 2023-03-28
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Make `core::task::Waker` FFI-safe, i.e. ensure that its layout is stable, that `core::task::RawWakerVTable`'s layout is stable, and that the function pointers in the v-table may be used with a stable calling convention.
+
+# Motivation
+[motivation]: #motivation
+
+Futures have now become a core feature of Rust, and returning boxed futures from trait-methods has become a very common pattern when working with `async`.
+
+However, due to Rust's unstable default ABI, trait-objects are never FFI-safe (note that this doesn't stop users trying). Multiple crates (such as [`abi_stable`](https://crates.io/crates/abi_stable) and [`stabby`](https://crates.io/crates/stabby)) attempt to solve this problem by providing proc-macros that generate stable v-tables for the traits they annotate.
```

Comment:
It's worth mentioning [async-ffi](https://crates.io/crates/async-ffi) which covers exactly the FFI-compatible waker scenario.

Reply:
Thanks, I wasn't aware of this one, I shall look it up this instant :)

Reply:
After checking it out, it uses the first technique I envisioned for `stabby`: allocating a new Waker on `poll`. `stabby` now only allocates on the first call to `waker.clone()`